src/transport/SessionManager.cpp

```diff
@@ -785,6 +806,26 @@ void SessionManager::ExpiryTimerCallback(System::Layer * layer, void * param)
     mgr->ScheduleExpiryTimer(); // re-schedule the oneshot timer
 }
 
+
+void SessionManager::ShiftToSession(const SessionHandle & handle)
+{
+    VerifyOrDie(handle->IsSecureSession());
+    SecureSession * session = handle->AsSecureSession();
+    VerifyOrDie(session->GetSecureSessionType() == SecureSession::Type::kCASE);
+    ScopedNodeId node = session->GetPeer();
+    mSecureSessions.ForEachSession([&](SecureSession * oldSession) {
+        if (session == oldSession)
+            return Loop::Continue;
+
+        if (oldSession->GetSecureSessionType() == SecureSession::Type::kCASE && oldSession->GetPeer() == node)
+        {
+            oldSession->TryShiftToSession(handle);
+        }
```

Comment:
The shared secret will differ every time a session is established, and so will the keys. I am unsure how this is meant to work. This would require comments in here to express why this is OK, and what assumptions are made.

Also, we need to make sure that the local node, fabricIndex and CATs match, as a validity check, since this API is public to SessionManager and could be alled (wrongly) outside of CASE/PASE session establishment

Reply:
They are inside `SecureSession::TryShiftToSession` now.